src/generators/java/JavaGenerator.ts

```diff
@@ -3,11 +3,13 @@ import {
   CommonGeneratorOptions,
   defaultGeneratorOptions,
 } from '../AbstractGenerator';
-import { CommonModel, CommonInputModel, RenderOutput } from '../../models';
-import { CommonNamingConvention, CommonNamingConventionImplementation, ModelKind, TypeHelpers } from '../../helpers';
+import { CommonModel, CommonInputModel, RenderOutput, OutputModel } from '../../models';
+import { CommonNamingConvention, CommonNamingConventionImplementation, ModelKind, TypeHelpers, FileHelpers, FormatHelpers } from '../../helpers';
 import { JavaPreset, JAVA_DEFAULT_PRESET } from './JavaPreset';
 import { ClassRenderer } from './renderers/ClassRenderer';
 import { EnumRenderer } from './renderers/EnumRenderer';
+import * as path from 'path';
```

Comment:
Did you try to generate the models on the website with this PR? I don't know how NextJS treats NodeJS related packages in the browser, but we should check it before merge to avoid broken rendering.

Reply:
`path` is okay, `fs` however, is a problem yes. Solvable in some cases - https://maikelveen.com/blog/how-to-solve-module-not-found-cant-resolve-fs-in-nextjs

Not sure where we want to draw the line? Should we just include `generateFullOutput` and don't help out with file creation? 🤔

Reply:
The idea with only `generateFullOutput` is nice. Hmm 🤔  Or maybe create separate class like `JavaFileGenerator` where you will have such a logic to generate and save to the files? Then everything will be tree-shakable, and NextJS should remove unused code in the website and other front-end tools, what do you think?

Reply:
Hmmm... Not sure whether the user experience is hurt by that. In most cases, I would not want to manually have to choose between the two 🤔 

What if we created a file interface that has to be provided to the function, so we dont include the `fs` module? 

i.e. we do something like:

```
export interface FileWriter {
  write(content: string, toFile: string): Promise<void>; 
}
  public async generateFile(input: Record<string, unknown> | CommonInputModel, outputDirectory: string, options: JavaRenderFullOptions, fileWriter: FileWriter): Promise<OutputModel[]> {
    let generatedModels = await this.generateFull(input, options);
    generatedModels = generatedModels.filter((outputModel) => { return outputModel.model.$id !== undefined; });
    for (const outputModel of generatedModels) {
      const filePath = path.resolve(outputDirectory, `${outputModel.modelName}.java`);
      await fileWriter.write(outputModel.result, filePath);
    }
    return generatedModels;
  }
```

Thoughts?

Reply:
But then someone has to implement that interface. With the new class I meant rather to create helper class which should have as argument to the constructor base generator of given language, so you will end with something like:

```ts
class JavaFileGenerator {
  constructor(
    private generator: JavaGenerator
  ) {}
  
  generateFile() { .... }
  ...
}
```

and then you have another module in which this fs would be separated and could be easily removed

Reply:
@jonaslagoni It's an another option, but then people will write such an code:

```js
async (content, file) => {await fs.promises.writeFile(file, content);}
```

every time. For me it makes more boilerplate than the second extra class :)

Also from our conversation in slack (https://asyncapi.slack.com/archives/CQVJXFNQL/p1633351369306600) you gave me one idea - we can create `FileGenerator` class for every generators, but it will work in this way that appropriate generator (e.g. JavaGenerator) will produce by some function (generateToFiles) array with context and `FileGenerator` will only render content to file and save it. What do you think? Do you understand my point of view?

Reply:
I decided to go with the inheritance option. Did not find a specific reason why you would need to have access to two generators 😄 

i.e. you use:

```ts
const generator = new JavaFileGenerator();
const outputFolder = './examples/generate-to-files/output';
const modelGenerationOptions = {
  packageName: 'TestPackageName'
};
const models = await generator.generateToSeparateFiles(jsonSchemaDraft7, outputFolder, modelGenerationOptions);
```

I really like this approach 😄